### PR TITLE
[components] Add ability to load directly from path

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

This allows us to use the config loading system to load components from a path directly. All `defs.yaml` files conform to the same schema of:

```yaml

component_type: <component type>

component_params: <component-specific params>

```

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26150?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26150** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26150?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26149** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26149?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26148** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26143** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26143?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26104** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26104?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/dagster/dagster_tests/components_tests/code_locations/python_script_location/scripts/defs.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+component_params:
+  component_type: python_script
+  components:
+    script_one:
+      - key: a
+      - key: b
+        deps: [up1, up2]
+    script_three:
+      - key: override_key
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/components_tests/code_locations/python_script_location/scripts/defs.yml:11]: Now that I see this I think it is important to restructure this schema to have an `asset_specs` key.

```yaml
component_type: collection
component_params:
  component_type: python_script
  components:
    script_one:
       assets:
          - key: a
          - key: b
             deps: [up1, up2]
    script_three:
       assets:
           - key: override_key
```

**Review by schrockn — CHANGES_REQUESTED** [review]
enough to do a spin

**Review by schrockn — APPROVED** [review]

**Comment by OwenKephart** [comment]
### Merge activity

* **Nov 27, 12:47 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/26150).

**Pull request merged**